[Extractors]: add guidance on YouTube session rate limit (#52)
yt-dlp/yt-dlp-wiki: Extractors.md

    --extractor-args "youtubetab:skip=webpage" --extractor-args "youtube:player_skip=webpage,configs;visitor_data=VISITOR_DATA_VALUE_HERE"

## Common YouTube Errors

#### `This content isn't available, try again later`

> This error is caused by your YouTube session (guest or account) hitting the YouTube video request rate limit. 

Reviewer: ```suggestion
This error is caused by your YouTube session (either guest or logged-in) exceeding the YouTube video request rate limit. 
```

> This error is caused by your YouTube session (either guest or logged-in) exceeding the YouTube video request rate limit. 

Reviewer: sorry for being pedantic, but maybe this makes more grammatical sense
```suggestion
This error is caused by your (guest or logged-in) YouTube session exceeding the YouTube video request rate limit. 
```

It is recommended to add a delay of around 5-10 seconds between downloads with `-t sleep` or [with the sleep options](https://github.com/yt-dlp/yt-dlp#workarounds).

> With the default yt-dlp settings, the rate limit for guest sessions is ~300 videos/hour. For account sessions, it is ~2000 videos/hour.

Reviewer: Maybe?
```suggestion
With the default yt-dlp settings, the rate limit for guest sessions is ~300 videos/hour (~1000 webpage/player requests per hour). For account sessions, it is ~2000 videos/hour (~4000 webpage/player requests per hour).
```

> With the default yt-dlp settings, the rate limit for guest sessions is ~300 videos/hour (~1000 webpage/player requests per hour). For account sessions, it is ~2000 videos/hour (~4000 webpage/player requests per hour).

Reviewer: to be consistent with above?
```suggestion
With the default yt-dlp settings, the rate limit for guest sessions is ~300 videos/hour (~1000 webpage/player requests per hour). For logged-in sessions, it is ~2000 videos/hour (~4000 webpage/player requests per hour).
```
or maybe "account" is more consistent with the other sections of this wiki page. I'd prefer them to at least be consistent within this section, but up to you

    --extractor-args "youtubetab:skip=webpage" --extractor-args "youtube:player_skip=webpage,configs;visitor_data=VISITOR_DATA_VALUE_HERE"

## Common YouTube Errors

#### `This content isn't available, try again later`

This error is caused by the YouTube session (guest or account) hitting the YouTube video request rate limit. With the default yt-dlp settings, the session rate limit is ~300 videos/hour. 
Note if using account cookies, the limit includes videos watched in the browser with that account.

It is recommended to add a delay of around 5-10 seconds between downloads with `-t sleep` or [with the sleep options](https://github.com/yt-dlp/yt-dlp#workarounds).

Other options include:

> - using only `web` client (`--extractor-args "youtube:player-client=web"`) which will increase the rate limit to a maximum of ~1000 videos/hour. Note that this client requires a [PO Token](https://github.com/yt-dlp/yt-dlp/wiki/PO-Token-Guide) to get working formats.

Reviewer 2: ```suggestion
- using only `web` client (`--extractor-args "youtube:player-client=web" -f "bv*+ba"`) will increase the rate limit to a maximum of ~1000 videos/hour. Note that this client requires a [PO Token](https://github.com/yt-dlp/yt-dlp/wiki/PO-Token-Guide) to get working formats. You can also use only the `tv` client (`--extractor-args "youtube:player-client=tv"`) which will let you download ~500 videos/hour but won't require a PoToken. **WARNING: premium formats (such as 616) are not available on these clients without YouTube Premium cookies. As such, downloads will likely be lower quality than with the default clients.**
```

(if I'm not mistaken, `-f "bv*+ba"` should force yt-dlp to error if there's no audio-only formats, preventing the user from accidentally downloading format 18)

Author: I think I'm going to keep this simple and remove the other options for now and just note to use a sleep. I might consider writing up a gist with all the advanced technical details later on and linking it here.
